### PR TITLE
[Scala 3] Add formatting for polymorphic functions

### PR DESCRIPTION
**Pull request by tgodzik**
Polymorphic types are a way to define a function parametrized by a type, complementary to `def method[T]`

More information here: http://dotty.epfl.ch/docs/reference/new-types/polymorphic-function-types.html

I was also looking at fold/unfold, but polymorphic function types cannot have curly braces around their body, so this will not make much sense.

### CODE REVIEW

File: scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala [hunk 1]
```diff
@@ -356,7 +356,7 @@ class Router(formatOps: FormatOps) {
 
       case FormatToken(T.RightArrow() | T.ContextArrow(), right, _)
           if leftOwner.is[Term.FunctionTerm] ||
-            (leftOwner.is[Template] &&
+            leftOwner.is[Term.PolyFunction] || (leftOwner.is[Template] &&
```
Comment (kitbellew) [inline scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala:359]: completely optional: the new term looks to go better, logically, with the previous line:
```
if leftOwner.is[Term.FunctionTerm] || leftOwner.is[Term.PolyFunction] ||
  (leftOwner.is[Template] && // unchanged
```
if it fits, of course.

File: scalafmt-core/shared/src/main/scala/org/scalafmt/util/TreeOps.scala [hunk 2]
```diff
@@ -341,7 +341,9 @@ object TreeOps {
           _: Type.Apply | _: Type.Param | _: Type.Tuple | _: Defn.Enum |
           _: Defn.EnumCase | _: Defn.ExtensionGroup =>
         true
-      case _: Term.FunctionTerm | _: Type.FunctionType => true
+      case _: Term.FunctionTerm | _: Term.PolyFunction | _: Type.FunctionType |
```
Comment (kitbellew) [inline scalafmt-core/shared/src/main/scala/org/scalafmt/util/TreeOps.scala:344]: also completely optional:
```
case _: Term.FunctionTerm | _: Type.FunctionType => true // unchanged
case _: Term.PolyFunction | _: Type.PolyFunction => true
```

**Review by kitbellew — APPROVED** [review]
ocd

File: scalafmt-core/shared/src/main/scala/org/scalafmt/util/TreeOps.scala (hunk 2 above)
Comment (tgodzik) [reply scalafmt-core/shared/src/main/scala/org/scalafmt/util/TreeOps.scala]: Changed!

File: scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala (hunk 1 above)
Comment (tgodzik) [reply scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala]: The if line is still to long to changed it to:
```scala
 if leftOwner.is[Term.FunctionTerm] ||
   leftOwner.is[Term.PolyFunction] ||      
  (leftOwner.is[Template] &&
```

**Pull request merged**